test/mailers/mailer_test.rb

```diff
@@ -0,0 +1,20 @@
+require "test_helper"
+
+class MailerTest < ActionMailer::TestCase
+  test "pass given block to #mail call" do
+    class TestMailer < Devise::Mailer
+      def confirmation_instructions(record, token, opts = {})
+        @token = token
+        devise_mail(record, :confirmation_instructions, opts) do |format|
+          format.html(content_transfer_encoding: "7bit")
+        end
+      end
+    end
+
+    Devise.mailer = TestMailer
```

Comment:
we can replace this setup with something like this:

```ruby
TestMailer.confirmation_instructions(create_user, 'confirmation-token').deliver_now
```

This way we don't change the global `Devise.mailer` or depend on `create_user` sending the email for us.